src/libraries/Microsoft.Bcl.Memory/src/System/Buffers/Text/Base64UrlDecoder.cs

```diff
@@ -0,0 +1,1343 @@
+                    if (status is not OperationStatus.InvalidData)
+                    {
+                        break;
+                    }
+
+                    source = source.Slice(localConsumed);
+                    bytes = bytes.Slice(localWritten);
+                }
+                while (!source.IsEmpty);
+
+                return status;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe int DecodeFourElements(byte* source, ref sbyte decodingMap)
+        {
+            // The 'source' span expected to have at least 4 elements, and the 'decodingMap' consists 256 sbytes
+            uint t0 = source[0];
+            uint t1 = source[1];
+            uint t2 = source[2];
+            uint t3 = source[3];
+
+            int i0 = Unsafe.Add(ref decodingMap, (int)t0);
+            int i1 = Unsafe.Add(ref decodingMap, (int)t1);
+            int i2 = Unsafe.Add(ref decodingMap, (int)t2);
+            int i3 = Unsafe.Add(ref decodingMap, (int)t3);
+
+            i0 <<= 18;
+            i1 <<= 12;
+            i2 <<= 6;
+
+            i0 |= i3;
+            i1 |= i2;
+
+            i0 |= i1;
+            return i0;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe void WriteThreeLowOrderBytes(byte* destination, int value)
+        {
+            destination[0] = (byte)(value >> 16);
+            destination[1] = (byte)(value >> 8);
+            destination[2] = (byte)value;
+        }
+
+        private static bool IsValidPadding(uint padChar) => padChar == EncodingPadEqual || padChar == EncodingPadPercentage;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int IndexOfAnyExceptWhiteSpace(ReadOnlySpan<byte> span)
+        {
+            for (int i = 0; i < span.Length; i++)
+            {
+                if (!IsWhiteSpace(span[i]))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsWhiteSpace(int value)
+        {
+            if (Environment.Is64BitProcess)
+            {
+                // For description see https://github.com/dotnet/runtime/blob/48e74187cb15386c29eedaa046a5ee2c7ddef161/src/libraries/Common/src/System/HexConverter.cs#L314-L330
+                // Lookup bit mask for "\t\n\r ".
+                const ulong MagicConstant = 0xC800010000000000UL;
+                ulong i = (uint)value - '\t';
+                ulong shift = MagicConstant << (int)i;
+                ulong mask = i - 64;
+                return (long)(shift & mask) < 0;
+            }
+
+            if (value < 32)
+            {
+                const int BitMask = (1 << (int)'\t') | (1 << (int)'\n') | (1 << (int)'\r');
+                return ((1 << value) & BitMask) != 0;
+            }
+
+            return value == 32;
```

Comment:
This method will only ever be used with byte / char inputs, right? i.e. we don't need to care about int values > ushort.MaxValue? If so, I believe you could use this implementation, which is the 32-bit version of the 64-bit version used, and it would be appropriate for both 32-bit and 64-bit processes:
```C#
static bool IsWhiteSpace(int value)
{
    Debug.Assert(value >= 0 && value <= ushort.MaxValue);
    uint charMinusLowUInt32;
    return (int)((0xC8000100U << (short)(charMinusLowUInt32 = (ushort)(value - '\t'))) & (charMinusLowUInt32 - 32)) < 0;
}
```

Reply:
Yes, it is only used for byte char inputs, thanks!